### DISCUSSION TITLE
Help Fleshing Out “How to Get Up & Running the Quick Start Way”

### DISCUSSION THREAD

**Discussion by paurray** [templaza/astroid-framework#559] [Q&A] [answered]
hello Astroid Community

First off the Astroid Quick Start Template 2 looks awesome. Thanks to @sonvnn 

**Background Info:**

I am on a mac and have to date being using MAMP.
Really dig MAMP particularly the versioning, clones etc.
But development currently moving at a snails pace :-(

**The Rabbit Hole:**

Whilst waiting on MAMP I went down a rabbit hole and installed:

https://herd.laravel.com/
https://dbngin.com/
https://www.phpmyadmin.net/downloads/

And of course downloaded:

https://github.com/templaza/astroid-framework/releases/tag/v3.0.9

(which I guess is using also Kickstart from Akeeba)
https://www.akeeba.com/products/akeeba-kickstart.html

Very roughly what I did probably/kind of messed up/missing stuff:

a) Installed the items of software listed above:
b) In DBning created a service and called it ASTROID
c) I made a custom watch folder for Herd but default should also work
d) I put the contents of the downloaded PHP package in a sub folder of the Herd Watch Folder
e) I put the contents of the Astroid Quickstart package in a sub folder of the Herd Watch Folder
f) I needed to edit the PHP config file to accept no password and link up with 127.0.0.1
g) What ever way this works I am then able to access the subfolders in the Herd Watch folder via a Web Browser
h) Created a DB in myPHP admin running from the URL served by Herd to Web Browser.
i) Access the Quickstart folder via URL served by Herd to Web Browser and perform install…
j) I can use the database name created previously and as I am doing this locally and switched of the password for the Database: root and no password seems to do the trick. Also instead of “localhost” I use 127.0.0.1

**Cutting to the chase:**

Long story short. Through a lot of trial and error I got it to work and am posting the video here. This is a draft version and I will remove towards the end of the month: https://youtu.be/9EkSf12unbM

It will be replaced with something a little slicker and more informative!

**My question:**

Am I technically doing the install correctly?

The whole interface here looks very much like Akeeba Kickstart right?!?

Is the error normal?

> You have not filled in all required fields (database host name, database name, database username, database password and table name prefix). This will most likely cause an error trying to connect to the database and/or the restored site.
> 
> Are you ABSOLUTELY SURE this is what you really meant to do?

Also

> “Server Specific Files warning”

Guess that this can also be ignored?

Am I missing something else?

People are I guess going to be using all kinds of different development environments so I guess it is important to just focus on the install itself any maybe other details there after...

Is there something I could potentially add to a future tutorial called eg:
“How to Get Up & Running the Quick Start Way”

thanks

Paul

**Comment by paurray** [comment]
@sonvnn
Also as stupid as this sounds when I try to access the back end in my case:
http://xjoomla-astroid-qs.test/administrator/index.php
I have no idea what user name password combination  to use? Is there a default password to access the Joomla/Astroid Quick Start  back end?

**Reply by sonvnn** [reply]
Hi @paurray 

At the moment I am using Akeeba for quickstart. It help me save time a lot. I will build a Joomla quickstart in the near feature when I have time. 

To learn more about install Akeeba Quickstart you can check this tutorial https://ltheme.com/install-quickstart-package-using-akeeba-kickstart/

Quickstart is running on Joomla 5 so you need check your server and make sure it follow the Joomla Technical Requirements https://manual.joomla.org/docs/next/get-started/technical-requirements/

Please let me know if you need further assistant. I just back from Lunar New Year Holiday and ready to answer your questions.

Thanks & Best Regards,
Sonny

**Comment by paurray** [comment]
Hi @sonvnn 

Ok I have it all working properly locally. And I understand the process almost….
With my local install (using Herd like an AMP) I can access the site set up/kickstart page 
I prepare things with: Astroid QuickStart Files in the right Folder and my Database all set up correctly locally!

Then all I have to do is click the Web Button next to the site I am setting up

eg

http://xjoomla-astroid-qs-tk3.test/

<img width="530" alt="JustClickWEBsymbol" src="https://github.com/templaza/astroid-framework/assets/9163424/55c1df18-328e-4f7d-a3c2-f671c87ef5e3">

and work through the steps…. All good.

What I do not understand is when installing the Quickstart Package via the cPanel with hosting or Server. The last step in the article you linked says use:

http://yourdomain.com/kickstart.php

I don’t see the kickstart.php in the astroid-quickstart package

What is my path if I am setting this up with live hosting/server?

thanks

Paul

**Reply by sonvnn** [reply]
You don't need kickstart.php. Just unzip package in the root folder -> go to your domain and follow instruction steps.

**Comment by paurray** [comment]
@sonvnn thanks for this

Hope to have a new video tutorial out in about a week. Article will look something like this:

- In a previous article “Astroid: Unleashed: Getting Up & Running the Traditional Way” we looked at getting up and running by installing the Astroid Framework and Templates on top of an existing Joomla install. At which point you have either already built your Joomla Site from scratch or you need to do so. In both scenarios you are going to have to do the leg work. Ensuring that the Content and Mechanics of Joomla work seamlessly with the Astroid Framework. The advantage of this approach is that you start with a blank canvas adding what you need when you need it. Very much a process of a process of addition rater than subtraction. Slowly getting your feet wet step by step. Whilst the disadvantage is that sooner rather than later you will have to perform a lot of very receptive work in Joomla, getting the site usable. As in creating menus, categories, articles, linking them up and all that Jazz. You might even hit a couple of road blocks this way that could potentially be avoided by taking the Quickstart approach.
- Which brings us to the idea of having a head start. And out of the gate having every thing working with Dummy Content and a Boilerplate Template. Enter the Quickstart Way. In this Video Tutorial we will be performing a brand new fresh install of Joomla in its entirety, with the Astroid Framework and a slick new Astroid Template all linked up with demo material. The Content and Mechanics of the site work seamlessly. Maybe not exactly as you would like it but it all works and looks great. This way you can potentially save yourself a ton of leg work. Using the Quick Start Way most of the work you will end up doing will be being modifying content and removing features that you do not require. This time around more of a process of subtraction than addition. The disadvantage of the Quick Start method is that it is not going to help much if you already have an existing site that technically works and is populated with piles of content. You could however also argue that having a poke around a fully functional Joomla/Astroid Site is a good way to understand how to build your own site from scratch. Or maybe you are a developer and want to use a workflow like this to get multiple sites up and running on a daily basis. In which case you will probably really dig this method.
- A couple of words of caution. This is currently all very new and things are changing all the time. First up you need to be able to run Joomla 5 to get this to work. Most decent Web Hoster's should be able to do this but you need to check just in case. The requirements for Joomla 5 are listed HERE. Alternatively you can do all this in a so called AMP. The process is very similar and this is actually the method I use for demonstration purposes in this tutorial. But not all AMPs are equal. Examples of AMPs include: MAMP, Wamp and XAMPP. I am on a Mac and my usual weapon of choice, MAMP is not Joomla 5 compatible. I ended up taking a slightly less trodded path and using Herd with DBning. It is early days and I can not really say if I can recommend this method or not. One last thing maybe you just want to do the install online and cut straight to the chase. And indeed in the spirit of unlocking the Power of the Astroid Quick Start package there are a lot of very good reasons to get this all up and running online from the start. My priority here is being able to illustrate the potential of the “Getting Up & Running the Quick Start Way” with the tools at my disposal. Your exact tools or combination of tools will very likely differ somewhat.

**Comment by paurray** [comment, marked as answer]
Article is now online here:
https://astroidframe.work/blog/astroid-hands-on/getting-up-running-the-quickstart-way
Related video is here:
https://www.youtube.com/watch?v=JaVMyr4ve24
Please feel free to watch, like, share, subscribe etc :-)

**Reply by sonvnn** [reply]
Great! Thank you @paurray 💯
